test/e2e/specs/metrics_server.go

```diff
@@ -0,0 +1,37 @@
+package specs
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/openshift-azure/test/sanity"
+)
+
+var _ = Describe("Openshift on Azure metrics server tests [MetricsServer][EveryPR]", func() {
+	It("should should return metrics for every node in the cluster [MetricsServer]", func() {
+		By("retrieving node metrics")
+		res, err1 := sanity.Checker.Client.Admin.MetricsServerV1beta1.NodeMetricses().List(metav1.ListOptions{})
+		nodes, err2 := sanity.Checker.Client.Admin.CoreV1.Nodes().List(metav1.ListOptions{})
+		Expect(err1).ToNot(HaveOccurred())
+		Expect(err2).ToNot(HaveOccurred())
+		Expect(res.Items).To(HaveLen(len(nodes.Items)))
+	})
+
+	It("should should return valid metrics for a given node [MetricsServer]", func() {
+		By("retrieving master node metrics")
+		res, err := sanity.Checker.Client.Admin.MetricsServerV1beta1.NodeMetricses().Get("master-000000", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res.Usage.Cpu().MilliValue() > 0)
+		Expect(res.Usage.Memory().Value() > 0)
+	})
+
+	It("should should return valid metrics for a given pod [MetricsServer]", func() {
+		By("retrieving metrics for pod master-api-master-000000")
+		res, err := sanity.Checker.Client.Admin.MetricsServerV1beta1.PodMetricses("kube-system").Get("master-api-master-000000", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res.Containers[0].Usage.Cpu().MilliValue() > 0)
+		Expect(res.Containers[0].Usage.Memory().Value() > 0)
+	})
```

Comment:
The e2e tests here appear to check that one can pull metrics with an admin config from the cluster, but I don't think that quite reflects the end-user behaviour we want to test; we want a customer admin to be able to deploy HPA. I'm fine with that test change coming in a follow-up PR, but I don't think this quite tests the use case we're building for yet.

Reply:
yes that's right. Customer admin do not have permission to list metrics or to deploy hpa as far as I can see, so I will have to add following rbac permissions to customer-admin-cluster (and -project?) and also change these tests. 
```
- apiGroups:
    - metrics.k8s.io
  resources:
    - pods
    - nodes
  verbs:
    - get
    - list
    - watch
```
and
```
- apiGroups:
  - autoscaling
  resources:
  - horizontalpodautoscalers
  verbs:
  - create
  - delete
  - deletecollection
  - get
  - list
  - patch
  - update
  - watch
```

Reply:
Without those RBAC changes, customers won't actually be able to use the metrics server, so let's proceed with adding them.

Reply:
@ehashman I've discovered that allowing `kubectl top pods` to customer admin would allow them to see pods that otherwise they cannot see. Maybe we can block it per namespace?
For now this is commit which will allow to query nodes usage, but not pods and deploy HPA: https://github.com/openshift/openshift-azure/pull/2099/commits/d9f7155de03e1d2d7d1d5901e77a1271ae215331

I also want to test now HPA, which I haven't done yet.

Reply:
> would allow them to see pods that otherwise they cannot see

Which pods? Stuff in kube-system or something?

I think it'd be ideal to figure out HPA now, since the reason we're adding metrics-server is to enable the HPA use case.

Reply:
I've tested the HPA (manually) - customer admin is able to deploy HPA and HPA scaling works too. As discussed offline, I will add a e2e test for deploying HPA as customer admin. 

For pods: Any namespace - as example bellow shows, the pod list is not available with `k get pods` but is available with `k top pods` (this is before I removed `pods` from the cluster role definition)

As discussed offline, we decided to enable to view `pods`, so I will add it back and update the tests accordingly.

```sh
[ openshift-azure (metrics-server-1) ] oc whoami                                                     
omirensk@rhcuppettgmail.onmicrosoft.com                                                                    
[ openshift-azure (metrics-server-1) ] k top pods -n openshift-azure-logging                            
NAME                                           CPU(cores)   MEMORY(bytes)
log-analytics-cluster-agent-5765c8dffb-p64fw   12m          95Mi  
log-analytics-node-agent-mvqn8                 6m           71Mi          
[ openshift-azure (metrics-server-1) ] k get pods -n openshift-azure-monitoring
No resources found.                                                                                                                                                                                             
Error from server (Forbidden): pods is forbidden: User "omirensk@rhcuppettgmail.onmicrosoft.com" cannot list pods in the namespace "openshift-azure-monitoring": no RBAC policy matched
```

Reply:
@ehashman 
the hpa section in the ClusterRole is not needed, customer admins are able to deploy HPAs without it in the non-openshift projects. Even in e2e test with `EndUser` the hpa seems to be deployed without any issue. So I didn't add another e2e for this bit. 

You can view the change in this commit: https://github.com/openshift/openshift-azure/pull/2099/commits/c71b95598b8bca2b61fa0fb0c5110531d7bf6d51

the following output demonstrates this for customer admin use-case:
```
$ oc whoami 
omirensk@rhcuppettgmail.onmicrosoft.com
$ k top pods
$ k top pods -n openshift-azure-logging
NAME                                           CPU(cores)   MEMORY(bytes)   
log-analytics-cluster-agent-67467699f5-989h9   16m          84Mi            
log-analytics-node-agent-mvqn8                 6m           79Mi            
$ 
$ k get hpa
NAME          REFERENCE                      TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
welcome-php   DeploymentConfig/welcome-php   <unknown>/80%   1         5         0          4m
$ k delete hpa welcome-php
horizontalpodautoscaler.autoscaling "welcome-php" deleted
$
$ k get hpa
No resources found.
$
$ k apply -f tmp/welcome-php-hpa.yaml 
horizontalpodautoscaler.autoscaling/welcome-php created
$
$ k get hpa
NAME          REFERENCE                      TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
welcome-php   DeploymentConfig/welcome-php   <unknown>/80%   1         5         0          4s
$
$ oc describe ClusterRole customer-admin-cluster
Name:                                           customer-admin-cluster
Created:                                        28 hours ago
Labels:                                         azure.openshift.io/owned-by-sync-pod=true
Annotations:                                    <none>
Verbs                                           Non-Resource URLs       Resource Names  API Groups                      Resources
[create delete get list patch update watch]     []                      []              [servicecatalog.k8s.io]         [clusterservicebrokers]
[create delete get list patch update watch]     []                      []              [ user.openshift.io]            [groups identities useridentitymappings users]
[create delete get list patch update watch]     []                      []              [ authorization.openshift.io]   [clusterrolebindings rolebindings]
[create delete get list patch update watch]     []                      []              [rbac.authorization.k8s.io]     [clusterrolebindings rolebindings]
[delete get list watch]                         []                      []              [ oauth.openshift.io]           [oauthclientauthorizations]
[create]                                        []                      []              [ authorization.openshift.io]   [resourceaccessreviews subjectaccessreviews]
[create]                                        []                      []              [authorization.k8s.io]          [subjectaccessreviews]
[create]                                        []                      []              [ project.openshift.io]         [projectrequests]
[get list watch]                                []                      []              []                              [events nodes persistentvolumes pod]
[create delete get list patch update watch]     []                      []              [ security.openshift.io]        [securitycontextconstraints]
[get list watch]                                []                      []              [ quota.openshift.io]           [clusterresourcequotas]
[get list]                                      []                      []              [ authorization.openshift.io]   [clusterpolicybindings]
[get list watch]                                []                      []              [ image.openshift.io]           [images imagestreamtags]
[get list update]                               []                      []              [ network.openshift.io]         [netnamespaces]
[get list]                                      []                      []              [ network.openshift.io]         [clusternetworks]
[get list watch]                                []                      []              [ build.openshift.io]           [buildconfigs builds]
[get list watch]                                []                      []              [metrics.k8s.io]                [nodes pods]
```